File: src/main/java/com/developer/drodriguez/model/Album.java
package com.developer.drodriguez.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document
public class Album implements Comparable<Album> {

    @Id
    private String id;
    private String name;
    private List<Song> songs;

    public Album(String name, List<Song> songs) {
        this.id = UUID.randomUUID().toString();
        this.name = name;
        this.songs = songs;
    }

    @Override
    public int compareTo(Album album) {
        return this.name.compareTo(album.name);
    }


}

File: src/main/kotlin/com/drodriguez/maestro/api/model/Album.kt
package com.drodriguez.maestro.api.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.util.UUID

@Document
data class Album(@Id val id: String = UUID.randomUUID().toString(), var name: String, var songs: List<Song>) : Comparable<Album> {
    override fun compareTo(album: Album): Int {
        return this.name.compareTo(album.name)
    }
}

